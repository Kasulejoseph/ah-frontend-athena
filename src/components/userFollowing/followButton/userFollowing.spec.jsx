import React from 'react';
import { shallow } from 'enzyme';
import Following from './userFollowing'

const props = {
  handleClick: jest.fn(),
  text: '',
  classValue: '',
  canFollow: false
}

describe('test follow button', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Following {...props} />)
  })
  it('should render follow button', ()=>{
    expect(wrapper).toMatchSnapshot()
  })
})