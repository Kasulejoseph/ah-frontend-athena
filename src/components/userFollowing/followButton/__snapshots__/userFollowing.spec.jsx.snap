// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test follow button should render follow button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <followButton
    canFollow={false}
    classValue=""
    handleClick={[MockFunction]}
    text=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "",
      "className": "",
      "onClick": [MockFunction],
      "type": "button",
    },
    "ref": null,
    "rendered": "",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "",
        "onClick": [MockFunction],
        "type": "button",
      },
      "ref": null,
      "rendered": "",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
